sdk/ai/azure-ai-projects/azure/ai/projects/aio/operations/_patch.py

```diff
@@ -0,0 +1,2472 @@
+         the run exceeds the number of
+         completion tokens specified, the run will end with status ``incomplete``. See
+         ``incomplete_details`` for more info. Default value is None.
+        :paramtype max_completion_tokens: int
+        :keyword truncation_strategy: The strategy to use for dropping messages as the context windows
+         moves forward. Default value is None.
+        :paramtype truncation_strategy: ~azure.ai.projects.models.TruncationObject
+        :keyword tool_choice: Controls whether or not and which tool is called by the model. Is one of
+         the following types: str, Union[str, "_models.AgentsApiToolChoiceOptionMode"],
+         AgentsNamedToolChoice Default value is None.
+        :paramtype tool_choice: str or str or
+         ~azure.ai.projects.models.AgentsApiToolChoiceOptionMode or
+         ~azure.ai.projects.models.AgentsNamedToolChoice
+        :keyword response_format: Specifies the format that the model must output. Is one of the
+         following types: str, Union[str, "_models.AgentsApiResponseFormatMode"],
+         AgentsApiResponseFormat Default value is None.
+        :paramtype response_format: str or str or
+         ~azure.ai.projects.models.AgentsApiResponseFormatMode or
+         ~azure.ai.projects.models.AgentsApiResponseFormat
+        :keyword metadata: A set of up to 16 key/value pairs that can be attached to an object, used
+         for storing additional information about that object in a structured format. Keys may be up to
+         64 characters in length and values may be up to 512 characters in length. Default value is
+         None.
+        :paramtype metadata: dict[str, str]
+        :keyword sleep_interval: The time in seconds to wait between polling the service for run status.
+            Default value is 1.
+        :paramtype sleep_interval: int
+        :return: AgentRunStream.  AgentRunStream is compatible with Iterable and supports streaming.
+        :rtype: ~azure.ai.projects.models.AsyncAgentRunStream
+        :raises ~azure.core.exceptions.HttpResponseError:
+        """
+        # Create and initiate the run with additional parameters
+        run = await self.create_run(
+            thread_id=thread_id,
+            assistant_id=assistant_id,
+            model=model,
+            instructions=instructions,
+            additional_instructions=additional_instructions,
+            additional_messages=additional_messages,
+            tools=tools,
+            temperature=temperature,
+            top_p=top_p,
+            max_prompt_tokens=max_prompt_tokens,
+            max_completion_tokens=max_completion_tokens,
+            truncation_strategy=truncation_strategy,
+            tool_choice=tool_choice,
+            response_format=response_format,
+            metadata=metadata,
+            **kwargs,
+        )
+
+        # Monitor and process the run status
+        while run.status in ["queued", "in_progress", "requires_action"]:
```

Comment:
Could this be refactored to return an azure.core.polling.LROPoller?

Reply:
A month ago, I discovered LROPoller in azure.core.   So I chat this with Jarno and Johan.   Long story short, at the end of the chat, I asked Johan,
"Am I correct that for example, we should have
create_vector_store_poll  calling create_vector_store following by calling get_vector_store , wait for number of seconds by parameter and poll again until status completed, then return the vector store?"
He said yes.
And we did chat about TypeSpec to generate LROPoller.

Reply:
Not blocking preview, but it would be nice to include a `timeout` param so a user could stop waiting if they wanted

Reply:
@kristapratico  per Howie below

Why I am not sure I can't resolve two comments from Krista, so I type here:

Krista: Not blocking preview, but it would be nice to include a timeout param so a user could stop waiting if they wanted
Howie: https://github.com/orgs/ai-platform-ml-platform/projects/7/views/1?pane=issue&itemId=86793660

Krista: Maybe i'm missing it, but I don't see the update. Was it only done with sync?
I think you refer to your previous comment related forcing to use keyword. I just added here
https://github.com/Azure/azure-sdk-for-python/pull/38426